### DISCUSSION TITLE
Unused pre_norm argument in ResnetBlock2D class

### DISCUSSION THREAD

**Discussion by masaishi** [huggingface/diffusers#6831] [General]
I've been exploring the `ResnetBlock2D` class within the `diffusers` library, specifically located at `diffusers/src/diffusers/models/resnet.py`, and I noticed something that might warrant a discussion or clarification.

The constructor of `ResnetBlock2D` has an argument `pre_norm` that is fixed to `True`:

```python
class ResnetBlock2D(nn.Module):
    def __init__(self, ..., pre_norm=True, ...):
        super().__init__()
        self.pre_norm = True
        # Other initialization code
```
https://github.com/huggingface/diffusers/blob/9cc59ba0891c3ccbebfcbb6784f4dde2d5bd8556/src/diffusers/models/resnet.py#L259C9-L259C29

I think the code should becomes `self.pre_norm = pre_norm`, and test result didn't change even if I change code.
If this modification is correct, I would like to send a pull request, but I am not familiar with the structure of models, and there is a possibility that it has some meaning and is fixed, so I posted this because I wanted to know about it.

**Comment by sayakpaul** [comment]
Cc: @yiyixuxu

**Comment by sayakpaul** [comment]
I think it could make sense default `pre_norm` to True and do what you proposed.

**Comment by masaishi** [comment]
Thank you for commenting. I will make a pull request tomorrow, because it can run test.
